### DISCUSSION TITLE
How to use moby for building container based systems using selected components.

### DISCUSSION THREAD

**Discussion by Om-Dodmani** [moby/moby#47397] [Q&A]
How to use moby for building container based systems using certain selected components. Is there a **demo** or some **configuration files** that I can run. Is there some reference config files that I can look at which explains the building of this system.

**Comment by AkihiroSuda** [comment]
- https://github.com/docker/docker-ce-packaging combines https://github.com/moby/moby , https://github.com/docker/cli , etc. to build Docker Engine
- https://github.com/balena-os/balena-engine forks https://github.com/moby/moby to build Balena Engine

**Comment by Om-Dodmani** [comment]
Do we need to create a dockerfile for specifying the components we need and
run it within moby repo?
Can you explain how to the moby repo structure is what is use of each
directory in moby project?
